Islandora/islandora
# D9 islandora

**Ticket**: https://github.com/Islandora/documentation/issues/1680
THIS PR depends on https://github.com/Islandora/islandora/pull/816

# What does this Pull Request do?

Updates Islandora Core for Drupal 9 support

# What's new?
* remove deprecated functions
* insert new functions
* use dependency injection
* clean up for php codesniffer
* updates travis to php 7.4
* updates context, migrate_source_csv, and flysystem to versions that supports d9
* temporarily point to crayfish-commons branch that supports d9 and php 7.4
* updates migration source config (keys => ids)
* includes an update hook for updating migration config

# How should this be tested?
Checkout the PR
Do a cache clear
Make sure you can do normal islandora stuff - make a node/media/file and have them all persist to fedora properly, get derivatives, etc.
Pass all the tests
Note that updatedb must be run
Due to dependency chain, I think this will only work on a d9 site

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.


# Interested parties
@Islandora/8-x-committers @kayakr @jordandukart
